provider/sidecar_test.go

```diff
@@ -101,13 +132,21 @@ func TestSidecar(t *testing.T) {
 
 			backs := prov.makeBackends(states)
 
```

Comment:
This seems like way too much stuff to be checking in a single test. The comments explaining all the checks are the giveaway. Even if you have to repeat the same loading of the data multiple times in separate tests, this should be broken into tests that are validating the same functionality. Just share the setup between the `Convey` blocks and give the `So` clauses a wrapper that explains what they're doing.